query.go

```diff
@@ -1096,6 +1098,18 @@ func (s *ChainService) sendTransaction(tx *wire.MsgTx, options ...QueryOption) e
 					response, sp.Addr(),
 				)
 				rejections[*broadcastErr]++
+
+				// To ever get (numReplied == len(rejections)), it is necessary
+				// to also increment numReplied. However, this would be a
+				// breaking change since common innocuous errors such as Mempool
+				// and Confirmed do not presently cause a non-nil error return.
```

Comment:
I think we should account for those cases then. Would you mind adding that? something like this:

```

func (c BroadcastErrorCode) IsCritical() bool {
	return c != Mempool && c != Confirmed	
}
```

Then, I think we can add the `numReplied++` here and then below we tally the rejections that are not critical and only return then we can keep the `if numReplied == numCriticalErrors` type logic

Reply:
ok i see now that the overall effect of the PR is just to close the quit channel and to not include these comments. 

So then rather: squash all the commits into 1 that just does that. I will open an issue to fix the other things you have pointed out. Nice catch!

Reply:
Right sorry for the confusion there.  I kept the commits "unsquashed" so you could see the issues illustrated.  Will squash it down to one commit.